src/macros.rs

```diff
@@ -129,47 +134,102 @@
 /// // Doesn't compile because `foo` isn't a `Bar`
 /// doc! { field!((foo in MyModel).(baz in Bar)): 0 };
 /// ```
+/// ```compile_fail
+///# use mongodm::mongo::bson::doc;
+///# use mongodm::field;
+///#
+///# struct MyModel {
+///#     foo: i64,
+///#     bar: Bar,
+///# }
+///#
+///# struct Bar {
+///#     baz: i64,
+///#     third: Third,
+///# }
+///#
+///# struct Third {
+///#     a: String,
+///# }
+///#
+///# doc! { field!((bar in MyModel).(third in Bar).@@(b in Third)): 0 };
```

Comment:
Add a comment for why it doesn't compile.
```suggestion
///# // Doesn't compile because `b` isn't a member of `Third`
///# doc! { field!((bar in MyModel).(third in Bar).@@(b in Third)): 0 };
```

Also, I'm not sure what @@ are doing at this place. Is that supposed to work? Is "bar.third.$$b" a thing with mongo?
If that's another reason why the test is compile_fail, it needs a separate test.

Reply:
It's indeed not supposed to be there, I messed up on that one. Gonna fix it